backend/routes/download.js

```diff
@@ -0,0 +1,259 @@
+import fs from 'fs'
+import path from 'path'
+import { promisify } from 'util'
+
+import { Router } from 'express'
+import JSZip from 'jszip'
+
+import Entry from '../models/entry'
+import Group from '../models/group'
+import Image from '../models/image'
+import Show from '../models/show'
+import User from '../models/user'
+import { IMAGE_ENTRY } from '../constants'
+import config from '../config'
+import sequelize from '../config/sequelize';
+
+const readFileAsync = promisify(fs.readFile)
+
+const IMAGE_DIR = config.get('upload:imageDir')
+
+const router = Router()
+
+router.route('/zips/:showId')
```

Comment:
Just a general comment on this function. This looks like the right approach to the problem, but I kind of want it to be split into a few smaller functions. I think it's fine and we don't need to change it now though.